test/e2e_federation/authn.go

```diff
@@ -35,16 +35,20 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 	f := fedframework.NewDefaultFederatedFramework("federation-apiserver-authn")
 
 	var _ = Describe("Federation API server authentication [NoCluster]", func() {
+		var (
+			clusters fedframework.ClusterSlice
+		)
 		BeforeEach(func() {
 			fedframework.SkipUnlessFederated(f.ClientSet)
+			clusters = f.GetRegisteredClusters()
 		})
 
 		It("should accept cluster resources when the client has certificate authentication credentials", func() {
 			fcs, err := federationClientSetWithCert()
 			framework.ExpectNoError(err)
 
 			nsName := f.FederationNamespace.Name
-			svc := createServiceOrFail(fcs, nsName, FederatedServiceName)
+			svc := createServiceOrFail(fcs, nsName, FederatedServiceName, clusters)
```

Comment:
in order to call 'clusters = f.GetRegisteredClusters()' inside `createServiceOrFail`, I need to access the instance of `f *framework.Framework`. 
Also `createServiceOrFail` is a generic function used in deferent test cases, that initialise differently the  `f *framework.Framework`. 

@shashidharatd, Do you prefer that it stay like it is, or change the function signature by replacing `clusters` by `f` ? or do you see another way to revert passing `clusters` from caller of `createServiceOrFail`.